department-of-veterans-affairs/abd-vro
# domain-cc pytest GH action

## What was the problem?
There was no action to run and display python unit tests for the domain-cc app in PRs

Associated tickets or Slack threads:

- https://github.com/department-of-veterans-affairs/abd-vro/issues/1638

## How does this fix it?[^1]

Adds a new GH Action step to the test-code workflow which runs `pytest` if there have been changes to the `cc-app` subproject which hosts the python source files

Also addresses a couple of bugs in the testing code and requirements

## How to test this PR
- Testing is automatically performed as this is the substance of the PR


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
